components/arm/fake/fake.go

```diff
@@ -125,6 +131,22 @@ type Arm struct {
 	model      referenceframe.Model
 }
 
+// UpdateAction helps hinting the reconfiguration process on what strategy to use given a modified config.
+// See config.UpdateActionType for more information.
+func (a *Arm) UpdateAction(c *config.Component) config.UpdateActionType {
+	var err error
+	if newCfg, ok := c.ConvertedAttributes.(*AttrConfig); ok {
+		armModel := newCfg.ArmModel
+		modelPath := newCfg.ModelFilePath
+		a.model, err = buildModel(armModel, modelPath, a.Name)
```

Comment:
We want to be careful here that `a.model` isn't changed further down the call stack if an `error` is surfaced. Maybe worth mentioning in the docs for this function.

Reply:
Specifically, `modelFromName` can possibly `nil` out your name, and result in `buildModel` returning the simplified version if the model is not supported and no path is provided. I don't think that's what a user would expect.

Reply:
spoke offline with William about these comments.
We came to the consensus that is a.model is nil aka err is not nil implies we fatal out to make sure the user knows they have improperly configured the fake arm.

Reply:
will instead use `panic`. this also closes the server and will let the user know they have improperly configured the arm.